### PR TITLE
TRITON-1969 Validate boot_modules paths do not contain either "." or ".."

### PR DESCRIPTION
**Pull request by kusor**
Make sure that we don't allow any characters that `path.resolve` could _transform_ into our boot module file being saved outside bootfsdir when booter writes module files to cache.

### CODE REVIEW

File: lib/endpoints/boot_params.js [hunk 1]
```diff
@@ -47,16 +48,24 @@ function validateBootModules(boot_modules) {
             continue;
         }
 
+        if (/[.]+/.test(path.dirname(bm.path))) {
```
Comment (mgerdts) [inline lib/endpoints/boot_params.js]: That also matches a component that looks like `...` or `some.dir`.  It also seems that the last component being a `.` or `..` should be invalid. Perhaps better as:

```
paths = path.split('/');
if (paths.indexOf('.') != -1 || paths.indexOf('..') != -1)
```

File: test/test-servers.js [hunk 2]
```diff
@@ -677,7 +677,20 @@ function testServerSysinfo(t) {
...
+            client.put('/boot/' + uuid, {
+                boot_modules: [ {
+                    path: './etc/ppt_aliases',
+                    content:
+                        new Buffer(4 * 1024).toString('base64')
+                }]
+            }, function _onPut(err, req, res) {
+                t.ok(err, 'expected path error');
+                t.notEqual(err.message.indexOf('Invalid path'),
+                    -1, 'Invalid path');
```
Comment (mgerdts) [inline test/test-servers.js]: a couple more tests should be added to cover valid components that have a `.` in them.

File: lib/endpoints/boot_params.js (hunk 1 above)
Comment (kusor) [reply lib/endpoints/boot_params.js]: I think I've got a better and simpler approach. Use `path.normalize` and compare against provided path. If those aren't exactly the same, complain. If for some reason that's not working correctly, fill a bug for nodejs ;-)

File: test/test-servers.js (hunk 2 above)
Comment (kusor) [reply test/test-servers.js]: I've added some fancy path test too, hope it's enough with that one.

File: lib/endpoints/boot_params.js (hunk 1 above)
Comment (kusor) [reply lib/endpoints/boot_params.js]: Oh, also note I've moved the whole validation function to `lib/validation/endpoints.js` which seems to be the correct place.

**Pull request merged**